backend/src/authentication/strategies/jwt.strategy.ts

```diff
@@ -4,26 +4,32 @@ import { Injectable } from '@nestjs/common';
 import { ConfigService } from '@nestjs/config';
 import { UserService } from '../../user/user.service';
 import { User } from '@prisma/client';
-import { EnvironmentVariableKeys } from '../../config/environment-variable-keys';
 import { TokenPayload } from '../types/token-payload.interface';
+import { ActiveProfilesService } from '../../global/active-profiles/active-profiles.service';
+import { EnvironmentVariableKeys } from '../../shared/constants/environment-variable-keys';
+import { testUser } from '../../global/test-data/user-test-data.service';
 
 @Injectable()
 export class JwtStrategy extends PassportStrategy(Strategy) {
   // tslint:disable:no-unused-variable
   constructor(
     private readonly configService: ConfigService,
+    private readonly activeProfilesService: ActiveProfilesService,
     private readonly userService: UserService
   ) {
     super({
       jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
-      secretOrKey: configService.get(
+      secretOrKey: configService.get<string>(
         EnvironmentVariableKeys.JWT_ACCESS_TOKEN_SECRET
       ),
     });
   }
 
   public async validate(payload: TokenPayload): Promise<User> {
-    // TODO: if test profile on, create test user
+    if (this.activeProfilesService.isTestProfileActive()) {
+      return this.userService.findByEmail(testUser.email);
+    }
```

Comment:
if the `test` profile is active, no need to manually log in. this way, tests can focus solely on what needs to be tested and not have to worry about logging in